crates/memory/src/process.rs

```diff
@@ -25,6 +25,14 @@ pub enum ModuleError {
     ListModules,
 }
 
+#[derive(Debug)]
+pub enum MemoryError {
+    InvalidParameters, // When some parameters are invalid
+    NullPointer,       // When a ptr is 0
+    Unset,             // When a value is unset (ie None)
+    ReadError,         // when memory cant be read
+}
+
 pub struct Error;
```

Comment:
Should this one still be here?

Reply:
ill try nuking it - i left it in case it was still used internally :)